server/public/model/config.go

```diff
@@ -4052,10 +4088,18 @@ func (s *DataRetentionSettings) isValid() *AppError {
 		return NewAppError("Config.IsValid", "model.config.is_valid.data_retention.message_retention_days_too_low.app_error", nil, "", http.StatusBadRequest)
 	}
 
+	if s.MessageRetentionHours != nil && *s.MessageRetentionHours < 0 {
```

Comment:
Yeah I like that idea, it makes it easier to read the configs.

Reply:
@vish9812 So I have updated the code to support setting `days` to `0` but I have a question about acceptable behaviour for customers who are upgrading. 

I updated the default value for `days` to `0` and the `hours` default is `8760`. This means new customers will use our new hours config by default. 
Customers who are upgrading will now have both `days` and `hours` set on startup but the start up will fail because of this new condition in `is_valid`. They will need to set their old `days` config to `0` or update the new `hours` config to `0`. 

I guess this is ok as long as the error message is descriptive and the changelog details the steps?

cc/ @hanzei

Reply:
Interesting, I wonder if this new `isValid` check should be considered a breaking change since the customers will _have to_ update the config.
And if they do then they might as well just use the new `Hours` settings and the old `Days` setting can simply be removed entirely.

Because ideally in a backwards-compatible change, they wouldn't have to change anything in their current system and would transition to the new setting as per their convenience and the new setting would just work.

So, we may need to revert the check to allow both days and hours simultaneously as a valid value.
And, if we are okay with them updating their old values while upgrading then we should just remove the whole `days` config.

Thoughts?

cc: @Willyfrog

Reply:
Yeah I guess this is considered a breaking change, should I just revert to allow both configs to be > 0?

What's your take @Willyfrog?

Reply:
Do we really want to allow both values to coexist? or should we move to the new hours retention?

If we move to the new one: we have done this kind of update in the past (like moving sessions from days to hours) and the process there was:
- if there is no days value, populate the default in hours
- if there is a days values, populate the new value with the old value transformed (days*24 in the example mentioned above) 
Is that an option we could consider?

In the case we want both values to coexist (what would be the reason for it?), I'd suggest the default value for hours to be 0 and keep the old default in days. That way there is no breaking change and to use the new functionality one has to understand what does it mean each option. Global retention of <1day is not a very common scenario, so I'd push to use it only those who need it (again if we are keeping both options)

Reply:
> if there is a days values, populate the new value with the old value transformed (days*24 in the example mentioned above)

The problem with this is that when we try to set the new value we don't have access to the old value if it's an env variable or if it's stored in the configurations table. We only have access to the config.json value unfortunately.

> In the case we want both values to coexist (what would be the reason for it?), I'd suggest the default value for hours to be 0 and keep the old default in days. That way there is no breaking change and to use the new functionality one has to understand what does it mean each option. Global retention of <1day is not a very common scenario, so I'd push to use it only those who need it (again if we are keeping both options)

I think I will revert the last commit and set the default value for hours to 0. I'll support both hours and days but if you set hours > 0 we will use the hours value. System console will now also always save to the hours field. @vish9812 @Willyfrog Are you ok with that approach?

Reply:
@BenCookie95 yeah, the revert SGTM.

Reply:
👍🏼

Reply:
Done, I discovered another bug for when you are using an env variable but that last commit is reverted